internal/storage/block_storage.go

```diff
@@ -490,14 +490,18 @@ func (b *BlockStorage) UpdateBalance(
 	}
 
 	var existingValue string
-	if exists {
+	switch {
```

Comment:
> When also using a genesis balance bootstrap, it appears that the bootstrap is overriden and only the balance from the genesis block is returned. Before investigating further, I'd like to confirm what should be the expected behavior here. FWIW considering one or the other higher priority feels like a better solution. The other solution might be adding both the balances, which seems confusing.

I believe the preference here should be:
```suggestion
	switch {
	case exists: // this could happen if balances are bootstrapped and should not be overridden
	case parentBlock != nil && change.Block.Hash == parentBlock.Hash:
	default:
	}
```